### DISCUSSION TITLE
Match-Case

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#9850] [General]
Does microPython support the Python 3.10 match-case statement?
If not, is it under consideration?

MM

**Comment by dlech** [comment]
There is a an open issue for this at #7847.

**Reply by mmliam** [reply]
Thanks dtech,
Looks like it's a no-go and given the discussion and the date on that issue, it may be a never-go.

**Reply by jimmo** [reply]
It's a there-are-lots-of-priorities-and-a-small-team-and-this-is-not-the-highest-but-it-is-a-very-neat-feature-go. :grin: 

Matt's summary on https://github.com/micropython/micropython/issues/7847 is very accurate -- it's going to require a lot of work. At the end of the day though, MicroPython's goal is to run Python code, and so if structural pattern matching becomes a really important part of writing idiomatic Python then we'll have to consider it. (The walrus operator and f-strings were good examples of things that got implemented for this reason, although both very simple in comparison)

Also if code that uses structural pattern matching ends up being more concise (and therefore generates less bytecode and uses less RAM) then that's another good reason to work on it.
